modules/core/src/encoder/field/PurposeRestrictionVectorEncoder.ts

```diff
@@ -44,8 +44,23 @@ export class PurposeRestrictionVectorEncoder {
 
           }
 
-          // either end of the loop or there's a gap greater than 1 number
-          if (i === len - 1 || vendors[i + 1] > vendorId + 1) {
+          // we know that `len` is greater than zero because we entered the loop
+          const lastVendorId = vendors[len - 1];
+          const gvlVendorIds = prVector.gvl.vendorIds;
+
+          const nextGvlVendor = (vendorId: number): number => {
+
+            while (++vendorId <= lastVendorId && !gvlVendorIds.has(vendorId)) {
+            }
+
+            return vendorId;
+
+          };
+
+          /**
+           * either end of the loop or there are consecutive GVL vendor IDs
```

Comment:
nit: a bit of a confusing comment: `there are consecutive GVL vendor IDs` should be `there are GVL vendor IDs before the next one`.